changelog.d/9739.feature

```diff
@@ -0,0 +1 @@
+Added experimental support to cache `/sync` responses with config key `experimental_features.sync_cache_timeout_ms` (in milliseconds).
```

Comment:
(Do we announce experimental settings like these? Is this even experimental?)

Reply:
Well yes, we announce experimental features. I don't think this should be one though. Experimental features are more appropriate when we are making changes to the Matrix protocol; this is all internal to Synapse. 

If it needs shaking out before releasing (and I'm not convinced it does), we can do that with test deployments using branches rather than with feature flags.